tools/mergers/mergers.go

```diff
@@ -0,0 +1,105 @@
+package mergers
+
+import (
+	"bufio"
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/google/go-github/v32/github"
+	"github.com/pkg/errors"
+
+	"github.com/openshift/enhancements/tools/util"
+)
+
+// New creates a new Stats implementation
+func NewStats(query *util.PullRequestQuery) *Stats {
+	return &Stats{
+		PullRequestQuery: query,
+		lgtmByUser:       make(map[string]int),
+		approveByUser:    make(map[string]int),
+	}
+}
+
+// Stats holds the overall stats gathered from the repo
+type Stats struct {
+	*util.PullRequestQuery
+
+	lgtmByUser    map[string]int
+	approveByUser map[string]int
+}
+
+type UserCommandCount struct {
+	user         string
+	commandCount int
+}
+
+func getUserCommandCount(commandByUser map[string]int) []UserCommandCount {
+	var ret []UserCommandCount
+
+	for user, count := range commandByUser {
+		ret = append(ret, UserCommandCount{user, count})
+	}
+
+	// sort in descending order
+	sort.Slice(ret, func(a, b int) bool {
+		return ret[a].commandCount > ret[b].commandCount
+	})
+
+	return ret
+}
+
+func (s *Stats) getUserLGTMCount() []UserCommandCount {
+	return getUserCommandCount(s.lgtmByUser)
+}
+
+func (s *Stats) getUserApproveCount() []UserCommandCount {
+	return getUserCommandCount(s.approveByUser)
+}
+
+// Process extracts the required information from a single PR
+func (s *Stats) Process(pr *github.PullRequest) error {
+	isMerged, err := s.IsMerged(pr)
+	if err != nil {
+		return errors.Wrap(err,
+			fmt.Sprintf("could not determine merged status of %s", *pr.HTMLURL))
+	}
+
+	// Ignore PRs which weren't merged
+	if !isMerged {
+		return nil
+	}
+
+	comments, err := s.GetIssueComments(pr)
+	if err != nil {
+		return errors.Wrap(err,
+			fmt.Sprintf("could not fetch issue comments on %s", *pr.HTMLURL))
+	}
+
+	for _, c := range comments {
+		user := *c.GetUser().Login
+		scanner := bufio.NewScanner(strings.NewReader(c.GetBody()))
+		for scanner.Scan() {
+			line := scanner.Text()
+			if strings.HasPrefix(line, "/lgtm") {
+				s.lgtmByUser[user]++
+			} else if strings.HasPrefix(line, "/approve") {
```

Comment:
This will miss approvals added via the GitHub UI instead of a comment, like I did on https://github.com/openshift/enhancements/pull/609

To catch those, you also need to look at the "state" field in the review comment. https://docs.github.com/en/rest/reference/pulls#list-reviews-for-a-pull-request

Reply:
Yeah, thanks - I actually noted that when I saw what you did :)